### PR TITLE
Add registration parameters for observations event

### PR DESCRIPTION
**Pull request by sbernard31**
Add the registration in parameters of observations events. This allow for example to easily send request when we receive a new notification.

### CODE REVIEW

File: leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/ObservationServiceImpl.java [hunk 1]
```diff
@@ -220,13 +220,13 @@ public void onNotification(Request coapRequest, Response coapResponse) {
...
-            try {
-                // get registration
-                Registration registration = registrationStore.getRegistration(observation.getRegistrationId());
-                if (registration == null)
-                    // TODO Should we clean registrationIDs maps ?
-                    return;
+            // get registration
+            Registration registration = registrationStore.getRegistration(observation.getRegistrationId());
+            if (registration == null)
+                // TODO Should we clean registrationIDs maps ?
```
Comment (jvermillard) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/ObservationServiceImpl.java]: we should log something no?

Comment (sbernard31) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/ObservationServiceImpl.java]: Yes, we should ! I will add this. ERROR ?

Comment (msangoi) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/ObservationServiceImpl.java]: Yes, I think ERROR is fine, this case should not happen.

**Comment by sbernard31** [comment]
Integrated in master (commit 450abdcbb0364a5862ba221767916e7039df393e)

**Pull request closed without merging**